dbms/src/Storages/Page/V3/spacemap/SpaceMapRBTree.cpp

```diff
@@ -622,7 +632,7 @@ std::pair<UInt64, UInt64> RBTreeSpaceMap::searchInsertOffset(size_t size)
             rb_node_remove(node, &rb_tree->root);
             rb_free_entry(rb_tree, entry);
             max_cap = biggest_cap;
-            return std::make_pair(offset, max_cap);
+            return std::make_tuple(offset, max_cap, offset == last_offset);
```

Comment:
If the total space is [1, 10000], and the free space is [1,100] [2000, 2100] [3000, 4000],  `offset` and  `last_offset` here are 3000, it should be `reuse` rather than `expansion` ?

Reply:
changed